### PR TITLE
Update and fix golangci-lint issues

### PR DESCRIPTION
**Pull request by raharper**

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -5,13 +5,71 @@ linters-settings:
     lines: 80
     statements: 60
 
+linters:
+  # please, do not use `enable-all`: it's deprecated and will be removed soon.
+  # inverted configuration with `enable-all` and `disable` is not scalable during updates of golangci-lint
```
Comment (smoser) [inline .golangci.yml:10]: well, the way to do it is just to pin the version of golangci-lint that you're using.
Which we are doing in c-i (`.github/workflows/lint.yml`).
So a "disable" approach does seem still workable.  The person who wants a new lint version has to change lint.yml and then would be forced to make a decision on things to skip at that point.

I don't have really strong feeling either way.

File: .golangci.yml [hunk 2]
```diff
@@ -5,13 +5,71 @@ linters-settings:
...
+  # - testpackage
+  # - wsl
+  # - goimports
+
+# golangci.com configuration
+# https://github.com/golangci/golangci/wiki/Configuration
+service:
+  golangci-lint-version: 1.30.x # use the fixed version to not introduce new linters unexpectedly
+  prepare:
+    - echo "I wish I could be prepared. But, nothing to do here for the moment"
```
Comment (smoser) [inline .golangci.yml:70]: How does this get the version of golangci-lint to read this file ? I guess you have to use docker for that (per "how to test") in your url.

**Review by smoser — COMMENTED** [review]
I'm guessing you came to this problem because the golangci-lint on your PATH was not 1.22.2. Thats reasonable...

We really do have to pin the version of golangci-lint that we use, otherwise you'll get different results than c-i if they're different.

So the fix here really needs to include some way for 'make lint' to run with the same version that will run on c-i.

After we've done that, the changes to golangci.yml are not as important because anyone who changes the version of golangci-lint will have to make c-i pass one way or another.


How about this change, which puts the definition of the golangci version in Makefile, adds a rule to install it, and then uses that (./tools/golangci-lint-v1.30.0) everywhere.

```diff
diff --git a/.github/workflows/lint.yml b/.github/workflows/lint.yml
index ed9268f..a38a76b 100644
--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -12,8 +12,7 @@ jobs:
       uses: actions/checkout@v1
     - name: Install golangci-lint
       run: |
-        curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.22.2
+        make golangci-lint
     - name: Run linters
       run: |
-        export PATH=$PATH:$(go env GOPATH)/bin   
-        golangci-lint run --enable-all ./...
\ No newline at end of file
+        make lint
diff --git a/.golangci.yml b/.golangci.yml
index 54ececb..d4a5449 100644
--- a/.golangci.yml
+++ b/.golangci.yml
@@ -34,7 +34,6 @@ linters:
     - nakedret
     - noctx
     - nolintlint
-    - revive
     - rowserrcheck
     - staticcheck
     - structcheck
diff --git a/Makefile b/Makefile
index f3ef588..31503c8 100644
--- a/Makefile
+++ b/Makefile
@@ -5,6 +5,10 @@ LDFLAGS := "${ldflags:+$ldflags }-X main.version=${ver}${suff}"
 BUILD_FLAGS := -ldflags "-X main.version=$(VERSION_FULL)"
 ENV_ROOT := $(shell [ "$$(id -u)" = "0" ] && echo env || echo sudo )
 
+## why did you pick this version?
+GOLANGCI_VER = v1.30.0
+GOLANGCI = ./tools/golangci-lint-$(GOLANGCI_VER)
+
 CMDS := demo/demo ptimg/ptimg
 
 GO_FILES := $(wildcard *.go)
@@ -32,10 +36,18 @@ gofmt: .gofmt
 	o=$$(gofmt -l -w .) && [ -z "$$o" ] || { echo "gofmt made changes: $$o"; exit 1; }
 	@touch $@
 
+
+golangci-lint: $(GOLANGCI)
+
+$(GOLANGCI):
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | \
+		sh -s -- -b $(dir $@) $(GOLANGCI_VER) || { rm -f $(dir $@)/golangci-lint; exit 1; }
+	mv $(dir $@)/golangci-lint $@
+
 lint: .lint
 
-.lint: $(ALL_GO_FILES)
-	golangci-lint run
+.lint: $(ALL_GO_FILES) $(GOLANGCI)
+	$(GOLANGCI) run ./...
 	@touch $@
 
 test:
@@ -59,4 +71,4 @@ debug:
 clean:
 	rm -f $(CMDS) coverage*.txt coverage*.html .lint .build
 
-.PHONY: debug check test test-all gofmt clean all lint build
+.PHONY: debug check test test-all gofmt clean all lint build golangci-lint
```

Last thing, curious. Why did you choose 1.30.x ?

File: .golangci.yml (hunk 1 above)
Comment (raharper) [reply .golangci.yml:10]: Me neither.  I've seen other projects use the opt-in approach so enable-all doesn't "break" thinks with newer version of golint-ci or their components.   You're quite right about pinning the version though.

**Comment by raharper** [comment]
> How about this change, which puts the definition of the golangci version in Makefile, adds a rule to install it, and then uses that (./tools/golangci-lint-v1.30.0) everywhere.

Sure, let me try pulling this in.

**Comment by smoser** [comment]
I added to this at https://github.com/anuvu/disko/pull/110.
I'm going to close this one, @raharper  please review the other.

**Pull request closed without merging**